### DISCUSSION TITLE
💡尝试 PaddleSpeech 处理 SSML 输入

### DISCUSSION THREAD

**Discussion by david-95** [PaddlePaddle/PaddleSpeech#2538] [Show and tell]
尝试 PaddleSpeech 处理 SSML 输入
参考：[SSML标记语言介绍](https://help.aliyun.com/document_detail/101645.html)
pr 链接：https://github.com/PaddlePaddle/PaddleSpeech/pull/2531

**目前只支持中文的多音字的处理**， 通过标签给字或词 指定拼音，以下为文本样例. 可以把下面的文本保存在 sentences.txt 中：

❣️使用中英文混合合成时，请不要使用 SSML 标签，因为目前的 SSML 标签判断是放在中文文本前端中的，中英文混合合成时会先对文本进行语言判断，SSML 标签会被分类成英文，无法正常使用

❣️标签跨多句，不太好处理，有可能截断，不成 xml 了，最好标签不跨句子。如果输入文本中包含 `<speak> </speak>` 标签，会走带 SSML 处理的前端，**不会按照标点进行切分，如果输入过长可能会 OOM**

```text
00001 "希腊"神话,不只是人类的想象!
00002 你好吗？<speak>前浪<say-as pinyin='dao3'>倒</say-as>在沙滩上,沙滩上倒了一堆<say-as pinyin='tu3'>土</say-as>。 想象<say-as pinyin='gan1 gan1'>干干</say-as>的树干倒了, 里面有个干尸，不知是被谁<say-as pinyin='gan4'>干</say-as>死的。</speak>你很好。哈哈哈
```

https://user-images.githubusercontent.com/15189190/196130004-6fc4722c-17c5-4ff0-b184-3c3cb0801fd7.mp4

https://user-images.githubusercontent.com/15189190/196130019-e2d46752-ca88-40e6-b474-48b0072d5902.mp4

解释：
- 用于标识拼音的文本需放在`<speak> </speak>`标签之内，且每个语音合成任务只能包含一个`<speak> </speak>`标签。
- 多音字 用 <say-as></say-as> 标签， 设置 `pinyin` 的属性
- 多个字的拼音需要用**空格**分开， 比如上面例子中的 干干 pinyin="gan1 gan1"

**Comment by yt605155624** [comment]
🎉🎉🎉 中英文混合语音合成前端已加入 SSML 功能，参考: 
- https://github.com/PaddlePaddle/PaddleSpeech/pull/2830
